File: doc/porting/CanTossFakeItem.cs
using System;

namespace Assets.Scripts.GamePlayScene.Mechanics.Items
{
	public class CanTossFakeItem : Item, ICanExplodeAtLeast2
	{
		public CanTossItem MasterCanToss;

		public override ItemType GetItemType()
		{
			return ItemType.CanToss;
		}

		public override void ChangeSorting(Sorting s)
		{
		}

		public override Sorting GetSorting()
		{
			return default(Sorting);
		}

		public override int GetScore()
		{
			return 250;
		}

		public override bool CanBeExplodedBySpecialItem()
		{
			return true;
		}

		public override bool CanBeTapped()
		{
			return true;
		}

		public override bool CanBeExplodedByBooster()
		{
			return true;
		}

		public override bool CanGenerateScoreOnEachTryExplode()
		{
			return true;
		}

		public override bool CanFall()
		{
			return false;
		}

		public override bool CanCastShadow()
		{
			return false;
		}

		public override void PlayWrongMove()
		{
			this.MasterCanToss.PlayWrongMove();
		}

		public override bool CanExplodeDueToExplodeInNeigbour(MatchType sourceType)
		{
			return true;
		}

		public override void StartCollect()
		{
			this.MasterCanToss.StartCollect();
		}

		public override bool TryExplode()
		{
			return this.MasterCanToss.ExplodeBy(this);
		}

		public override void StartExplode()
		{
			this.MasterCanToss.StartExplode();
		}

		public void CheckIfExtraExplodeNeeded()
		{
			this.MasterCanToss.CheckIfExtraExplodeNeeded();
		}

		public void MarkShouldExplodeAtLeast2()
		{
			this.MasterCanToss.MarkShouldExplodeAtLeast2();
		}
	}
}

File: classes/game/item/CanTossFakeItem.cpp
#include "CanTossFakeItem.h"
#include "CanTossItem.h"
#include "../Animation.h"
#include "global/SoundManager.h"


CanTossFakeItem::CanTossFakeItem() {
}

CanTossFakeItem::~CanTossFakeItem() {
}


void CanTossFakeItem::PlayWrongMove() {
	this->MasterCanToss->PlayWrongMove(); 
}

void CanTossFakeItem::StartCollect() {
	this->MasterCanToss->StartCollect();
}

bool CanTossFakeItem::TryExplode() {
	return this->MasterCanToss->ExplodeBy(this);
}

void CanTossFakeItem::StartExplode() {
	this->MasterCanToss->StartExplode();
}

/// non override
void CanTossFakeItem::CheckIfExtraExplodeNeeded() {
	this->MasterCanToss->CheckIfExtraExplodeNeeded();
}

void CanTossFakeItem::MarkShouldExplodeAtLeast2() {
	this->MasterCanToss->MarkShouldExplodeAtLeast2();
}